scripts/migrate-io/migrate-consul.js

```diff
@@ -10,18 +10,380 @@ const {
+      image: '/consul/img/og-image.png',
+      icon: [{ href: '/consul/_favicon.ico' }],
+    },
+    alertBannerActive: false,
+    alertBanner: evalDataFile(path.join(repoDirs.data, 'alert-banner.js')),
+    version: evalDataFile(path.join(repoDirs.data, 'version.js')),
+    subnavItems: [],
+  }
+
+  // write product data to file
+  await exec(`rm -rf ./src/data/${productData.slug}.json`)
+  fs.writeFileSync(
+    `./src/data/${productData.slug}.json`,
+    JSON.stringify(productData, null, 2) + '\n',
+    'utf8'
+  )
+
+  //
+  // PAGES FOLDER SETUP
+  //
+  // delete some page files we don't need
+  const filesToDelete = [
+    '_app.js',
+    '_document.js',
+    '_error.jsx',
+    'jsconfig.json',
+    '404.jsx',
+    'print.css',
+    'style.css',
+    'index.jsx',
+  ]
+  for (let i = 0; i < filesToDelete.length; i++) {
+    const filepath = path.join(destDirs.pages, filesToDelete[i])
+    await exec(`rm -f ${filepath}`)
+  }
+  //
+  // DEPENDENCIES
+  //
+  // install dependencies required for proxied pages
+  const missingDependencies = [
+    '@hashicorp/react-text-splits',
+    '@hashicorp/react-featured-slider',
+  ]
+  console.log('⏳ Installing dependencies...')
+  // await exec(`npm i ${missingDependencies.join(' ')}`)
+  console.log('✅ Done')
+  //
+  // LIB FILES
+  //
+
+  //
+  // COMPONENTS
+  //
+  // copy components into dedicated directory
+  const missingComponents = [
+    'subnav',
+    'footer',
+    'homepage-hero',
+    'consul-on-kubernetes-hero',
+    'features-list',
+    'block-list',
+    'side-by-side',
+    'card-list',
+    'docs-list',
+    'downloads-props',
+  ]
+  for (let i = 0; i < missingComponents.length; i++) {
+    const srcPath = `${repoDirs.components}/${missingComponents[i]}`
+    const destPath = `${destDirs.components}/${missingComponents[i]}`
+    await exec(`cp -r ${srcPath}/ ${destPath}`)
+  }
+  await editFile(
+    `${destDirs.components}/downloads-props/index.jsx`,
+    (contents) => {
+      return contents.replace(
+        `'../../pages/downloads/style.module.css'`,
+        `'pages/_proxied-dot-io/consul/downloads/style.module.css'`
+      )
+    }
+  )
+
+  await patchSubnav(`${destDirs.components}/subnav/index.jsx`)
+  // home page components
+  const componentsThatNeedImportPathUpdates = []
+  await Promise.all(
+    componentsThatNeedImportPathUpdates.map(async (componentPath) => {
+      const filePath = `${destDirs.components}/${componentPath}`
+      await editFile(filePath, (contents) => {
+        return contents.replace(
+          /from 'components\//g,
+          `from 'components/_proxied-dot-io/${productData.slug}/`
+        )
+      })
+      return true
+    })
+  )
+  //
+  // ASSETS
+  //
+  const assetsToCopy = [
+    // meta images
+    // '/img/favicons/favicon.ico', NOTE: actually a Consul favicon
+    // Favicon has been manually added as _consul-favicon.ico in public folder
```

Comment:
I'm not seeing a favicon for consul on this branch. (Not seeing one for Vault either 👀)

Reply:
I'll fix the Vault one in a separate PR!